cranelift/codegen/src/isa/aarch64/inst/mod.rs

```diff
@@ -717,9 +717,14 @@ pub enum Inst {
 
     /// FPU move. Note that this is distinct from a vector-register
     /// move; moving just 64 bits seems to be significantly faster.
+    /// `is_zeroing` gives a hint to the register allocator that we rely
```

Comment:
I think I'd prefer to use the word "hint" here, since it seems that this is a correctness requirement (right?) -- eliding a move that actually serves a zero-extending purpose would result in bad results.

Reply:
Alternately, could we turn the zeroing variant into an `Extend` variant, since that's functionally what it's doing?

Reply:
If it does become an Extend, probably better if https://github.com/bytecodealliance/wasmtime/pull/2414 goes in first.

Reply:
@cfallin I decided to add an `Inst::FpuExtend` variant because it felt cleaner.

Reply:
This looks better for sure; thanks!